### PR TITLE
🖍  Bento Lightbox: Customizable styles

### PR DESCRIPTION
**Pull request by caroqliu**
Partial for #31052

### CODE REVIEW

File: extensions/amp-lightbox/1.0/amp-lightbox.css [hunk 1]
```diff
@@ -18,4 +18,18 @@ amp-lightbox {
   width: 100%;
   height: 100%;
   position: fixed;
+  visibility: hidden;
```
Comment (dvoytenko) [inline extensions/amp-lightbox/1.0/amp-lightbox.css:21]: Let's just a comment explaining why we have `visibility:hidden` - e.g. it's canceled by the lightbox itself.

File: extensions/amp-lightbox/1.0/amp-lightbox.css [hunk 2]
```diff
@@ -18,4 +18,18 @@ amp-lightbox {
...
+  box-sizing: border-box;
+  background-color: rgba(0, 0, 0, 0.9);
```
Comment (dvoytenko) [inline extensions/amp-lightbox/1.0/amp-lightbox.css:23]: Weird that we have a background color with transparency. Are we ok with some very bright content seen through?

File: extensions/amp-lightbox/1.0/amp-lightbox.css [hunk 3]
```diff
@@ -18,4 +18,18 @@ amp-lightbox {
   width: 100%;
   height: 100%;
   position: fixed;
+  visibility: hidden;
+  box-sizing: border-box;
```
Comment (dvoytenko) [inline extensions/amp-lightbox/1.0/amp-lightbox.css:22]: Let's also add `box-sizing: border-box` to the JSS file. It's necessary on Preact level as well.

File: extensions/amp-lightbox/1.0/lightbox.js [hunk 4]
```diff
@@ -164,15 +163,15 @@ function LightboxWithRef(
...
           lightboxRef.current = r;
         }}
-        style={{visibility: 'hidden'}}
         size={true}
         layout={true}
         paint={true}
-        className={classes.lightboxContainWrapper}
+        part="lightbox"
+        wrapperClassName={`${classes.defaultStyles} ${classes.wrapper}`}
+        wrapperStyle={{visibility: 'hidden'}}
```
Comment (dvoytenko) [inline extensions/amp-lightbox/1.0/lightbox.js:174]: FYI: this is not great. But I guess we will fix in a followup PRs. Ideally we always set "final" values on the rendering level and modify whatever is needed for animation during the animation side effect.

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: extensions/amp-lightbox/1.0/amp-lightbox.css (hunk 2 above)
Comment (caroqliu) [reply extensions/amp-lightbox/1.0/amp-lightbox.css:23]: I think so, given publishers can change this now. It should be somewhat translucent so end users know there is more content on the page than the lightbox modal they have open - but maybe not necessary if we do not allow it to be open on load. I was actually thinking 0.9 might be too opaque.

File: extensions/amp-lightbox/1.0/amp-lightbox.css (hunk 1 above)
Comment (caroqliu) [reply extensions/amp-lightbox/1.0/amp-lightbox.css:21]: Done

File: extensions/amp-lightbox/1.0/lightbox.js (hunk 4 above)
Comment (caroqliu) [reply extensions/amp-lightbox/1.0/lightbox.js:174]: Moved to jss wrapper class - verified this is overridden by inline styles applied during animations.

Comment (dvoytenko) [reply extensions/amp-lightbox/1.0/lightbox.js:174]: Still seeing it here. But what I actually want is a little different:

1. No `wrapperStyle={{visibility: hidden}}`.
2. No `visibility:hidden` (or opposite, only `visibility:visible`) in JSS.
3. All animation nuances are applied within the corresponding `useLayoutEffect`. E.g. the opening animation starts by setting style to `visibility:hidden` and finishes by setting it to `visible`.

Basically, if we disable animation effect - the styles are fully correct from the start.

**Review by dvoytenko — APPROVED** [review]
LGTM with one ask.

File: extensions/amp-lightbox/1.0/lightbox.js (hunk 4 above)
Comment (caroqliu) [reply extensions/amp-lightbox/1.0/lightbox.js:174]: Done

**Pull request merged**